### PR TITLE
Cleanup dyn_conf

### PR DESCRIPTION
**Pull request by shailesh33**
o Mainly try to keep dyn_conf.h clean so its understood whats the
interface
    o Create a new enum type of secure server option and use that
    o reduce the warnings in dyn_conf.c
    o Get ride of ctx_id

Sorry about the messy diff. Its only because of function reordering

### CODE REVIEW

File: src/dyn_conf.h [hunk 1]
```diff
@@ -126,7 +71,7 @@ struct conf_pool {
     int                server_connections;    /* server_connections: */
     msec_t             server_retry_timeout_ms;  /* server_retry_timeout: in msec */
     int                server_failure_limit;  /* server_failure_limit: */
-    struct array       server;                /* servers: conf_server array */
+    struct conf_server *conf_datastore;
```
Comment (ipapapa) [inline src/dyn_conf.h]: Add comment about the conf_datastore similar to the rest.

File: src/dyn_core.c [hunk 2]
```diff
@@ -66,17 +63,18 @@ core_ctx_create(struct instance *nci)
 	/* initialize server pool from configuration */
 	status = server_pool_init(&ctx->pool, &ctx->cf->pool, ctx);
 	if (status != DN_OK) {
-		loga("Failed to initialize server pool!!!");
+		loga("Failed to initialize server pool!!! ... uninitializing");
```
Comment (ipapapa) [inline src/dyn_core.c]: We are unitializing every time status!=DN_OK in dyn_core. Let us keep the loga consistent.

File: src/dyn_server.c [hunk 3]
```diff
@@ -114,65 +114,10 @@ server_active(struct conn *conn)
...
-	struct server_pool *sp = data;
-
-	s->owner = sp;
-
-	return DN_OK;
-}
-
 static void
-server_deinit(struct array *server)
+server_deinit(struct server **pdatastore)
```
Comment (ipapapa) [inline src/dyn_server.c]: so we do not use server_deinit anymore?

File: src/dyn_client.c [hunk 4]
```diff
@@ -638,7 +638,7 @@ local_req_forward(struct context *ctx, struct conn *c_conn, struct msg *msg,
 
     conn_enqueue_inq(ctx, s_conn, msg);
     req_forward_stats(ctx, s_conn->owner, msg);
-    if(msg->data_store==DATA_REDIS){
+    if(g_data_store == DATA_REDIS){
```
Comment (ipapapa) [inline src/dyn_client.c]: why did we need this change?

File: src/dyn_server.c (hunk 3 above)
Comment (shailesh33) [inline src/dyn_server.c]: I did the right thing (almost)

File: src/dyn_core.h [hunk 5]
```diff
@@ -250,15 +236,14 @@ struct server {
...
-    uint32_t           idx;                  /* pool index */
     struct context     *ctx;                 /* owner context */
     struct conf_pool   *conf_pool;           /* back reference to conf_pool */
 
     struct conn        *p_conn;              /* proxy connection (listener) */
     uint32_t           dn_conn_q;            /* # client connection */
     struct conn_tqh    c_conn_q;             /* client connection q */
 
-    struct array       server;               /* server[] */
+    struct server      *datastore;               /* server[] */
```
Comment (ipapapa) [inline src/dyn_core.h]: If we change the name from server->datastore, then why we do not change the struct name as well?

File: src/dyn_client.c (hunk 4 above)
Comment (shailesh33) [inline src/dyn_client.c]: we do not have multiple datastores. So no point in keeping that in every connection and message.

Comment (shailesh33) [inline src/dyn_client.c]: I am trying to make the connection and message more sleek

File: src/dyn_dnode_msg.c [hunk 6]
```diff
@@ -31,7 +31,7 @@ dyn_parse_core(struct msg *r)
 {
    struct dmsg *dmsg;
    struct mbuf *b;
-   uint8_t *p, *token;
+   uint8_t *p = r->pos, *token;
```
Comment (ipapapa) [inline src/dyn_dnode_msg.c]: what was the issue before?

File: src/dyn_core.h (hunk 5 above)
Comment (shailesh33) [inline src/dyn_core.h]: that will be the next step yanni. Thanks for looking so closely. Today server and peer are the same structs. That should not be the case. but I guessed that change would be more exhaustive

File: src/dyn_dnode_msg.c (hunk 6 above)
Comment (shailesh33) [inline src/dyn_dnode_msg.c]: an uninitialized variable. thats it.

File: src/dyn_core.h (hunk 5 above)
Comment (ipapapa) [inline src/dyn_core.h]: so are we leaving this change for the future?

**Pull request merged**